utils.go

```diff
@@ -35,3 +48,87 @@ func getFileContents(file string) (string, error) {
 
 	return string(bytes), nil
 }
+
+func getKernelVersion() (string, error) {
+	contents, err := getFileContents(procVersion)
+	if err != nil {
+		return "", err
+	}
+
+	fields := strings.Fields(contents)
+
+	if len(fields) < 3 {
+		return "", fmt.Errorf("unexpected contents in %v", procVersion)
+	}
+
+	version := fields[2]
+
+	if version == "" {
+		return "", fmt.Errorf("unexpected contents in %v", procVersion)
+	}
+
+	return version, nil
+}
+
+func getDistroDetails() (name, version string, err error) {
+	files := []string{osRelease, osReleaseClr}
+
+	for _, file := range files {
+
+		contents, err := getFileContents(file)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+
+			return "", "", err
+		}
+
+		lines := strings.Split(contents, "\n")
```

Comment:
Something you may be interested in (not necessarily for this PR) is [bufio.Scanner](https://golang.org/pkg/bufio/#Scanner), that would avoid copying the file content a few times (getFileContents, then Split) and directly uses a io.Reader. It's also easier to unit test because one can craft io.Readers backed up by a in-memory slice, say [`strings.Reader`](https://golang.org/pkg/strings/#NewReader)

	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		fmt.Println(scanner.Text()) // Println will add back the final '\n'
	}